### PR TITLE
2 step login + authentication 

### PR DESCRIPTION
**Pull request by frdomovic**

### CODE REVIEW

File: buildPackage.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#delete lib folder and node_modules from react project 
+rm -rf lib && cd examples/calimero-login && rm -rf node_modules && \
```
Comment (xilosada) [inline buildPackage.sh:2]: this will fail if lib doesn't exist, isn't it?

File: examples/calimero-login/src/calimeroSdk.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { CalimeroSdk } from "calimero-auth-sdk";
 
 export default CalimeroSdk.init({
-    shardId: "iso-calimero-testnet",
+    shardId: "90-calimero-testnet",
```
Comment (xilosada) [inline examples/calimero-login/src/calimeroSdk.js]: ok, for development but we have to use envvars here

File: src/index.ts [hunk 3]
```diff
@@ -223,12 +78,42 @@ export class CalimeroSdk {
...
+      }
+
+      localStorage.setItem(AUTH_TOKEN_KEY,
+        authToken);
+      localStorage.setItem(MESSAGE_KEY,
+        message);
+      localStorage.setItem(ACCOUNT_ID,
+        accountId);
+      localStorage.setItem(PUBLIC_KEY,
+        JSON.stringify(publicKey));
```
Comment (xilosada) [inline src/index.ts]: confirmSignIn after the process

File: src/index.ts [hunk 4]
```diff
@@ -223,12 +78,42 @@ export class CalimeroSdk {
     clearLocalStorage();
     window.location.href = "/";
   }
+
+  setCredentials = () =>  {
```
Comment (xilosada) [inline src/index.ts]: try-catch?

Comment (xilosada) [reply src/index.ts]: Return a result

File: src/index.ts [hunk 5]
```diff
@@ -1,105 +1,20 @@
...
 
-
-const MESSAGE_KEY = "calimeroSecret";
-const MESSAGE_HASH_KEY = "calimeroSecretHash";
 const AUTH_TOKEN_KEY = "calimeroToken";
-const CALIMERO_TOKEN_KEY = "caliToken";
-//Max valid period for a token would be 30 days
-export const MAX_CALIMERO_TOKEN_DURATION = 1000 * 60 * 60 * 24 * 30;
-
-export class WalletData {
```
Comment (xilosada) [inline src/index.ts:15]: I wouldn't remove this but move it to another module. Dapps should be able to validate the token, also this library can be used in our backends.

**Review by xilosada — CHANGES_REQUESTED** [review]
great first iteration

File: src/index.ts [hunk 6]
```diff
@@ -126,87 +41,27 @@ export class CalimeroSdk {
...
-      console.error(err);
-      return false;
-    });
-  }
+  signIn = () => (!localStorage.getItem(MESSAGE_KEY)) && this.signMessage();
 
   signTransaction = (transactionString: string, callbackUrl: string) => {
     if(!this.isSignedIn) {
-      console.log("SignIn required before sign a transaction");
+      window.alert("SignIn required before sign a transaction");
```
Comment (xilosada) [inline src/index.ts]: Too aggressive, return a result. Dapps should manage the output of this function. Alert is ok for the example dapp

File: buildPackage.sh (hunk 1 above)
Comment (frdomovic) [reply buildPackage.sh:2]: It works even when its deleted

File: src/index.ts (hunk 5 above)
Comment (frdomovic) [reply src/index.ts:15]: Should then all passed data from the url hash (#) be also saved in local storage under calimeroSecret and walletData ? Currently I only take 4 parameters from it..

File: examples/calimero-login/src/calimeroSdk.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
 
 export default CalimeroSdk.init({
-    shardId: "slim-shardy-calimero-testnet",
-    walletUrl: "https://localhost:1234",
-    calimeroUrl: "https://api.development.calimero.network",
-    calimeroWebSdkService: "https://api.development.calimero.network",
+    shardId: process.env.REACT_APP_CALIMERO_SHARD_ID,
+    walletUrl: process.env.REACT_APP_WALLET_ENDPOINT_URL,
+    calimeroUrl: process.env.REACT_APP_CALIMERO_ENDPOINT_URL,
+    calimeroWebSdkService: process.env.REACT_APP_ALIMERO_WEB_SDK_SERVICE_URL,
```
Comment (xilosada) [inline examples/calimero-login/src/calimeroSdk.js]:
```suggestion
    calimeroWebSdkService: process.env.REACT_APP_CALIMERO_WEB_SDK_SERVICE_URL,
```
do we really need these 2

File: src/index.ts [hunk 8]
```diff
@@ -1,105 +1,20 @@
...
-    shardId: string,
-    from: Date = new Date(Date.now()),
-    to: Date = new Date(Date.now() + MAX_CALIMERO_TOKEN_DURATION))
-  {
-    this.accountId = accountId;
-    this.shardId = shardId;
-    this.from = from;
-    this.to = to;
-  }
+const MESSAGE_KEY = "calimeroSessage";
```
Comment (xilosada) [inline src/index.ts]:
```suggestion
const MESSAGE_KEY = "calimeroMessage";
```

File: src/index.ts [hunk 9]
```diff
@@ -223,12 +77,50 @@ export class CalimeroSdk {
...
+        } else if (error instanceof Error) {
+          return {error: error.message}
+        }
+      } 
+    }
+  }
+
+  confirmSignIn = () => {
+    if(window.location.hash){
+      window.location.replace(window.location.origin)
```
Comment (xilosada) [inline src/index.ts]: what about other query params?

**Review by xilosada — APPROVED** [review]

File: examples/calimero-login/src/calimeroSdk.js (hunk 7 above)
Comment (frdomovic) [reply examples/calimero-login/src/calimeroSdk.js:7]: websdkservice not used anywhere so gonna remove it

File: src/index.ts (hunk 9 above)
Comment (frdomovic) [reply src/index.ts]: window.location.replace(window.location.origin+window.location.pathname);
Should cover it 
 source: https://ui.dev/get-current-url-javascript

**Pull request merged**